### PR TITLE
add ArrayCache

### PR DESCRIPTION
**Pull request by veolys-nanne**
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/2343
https://github.com/open-orchestra/open-orchestra-base-api-bundle/pull/122
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/419

### CODE REVIEW

File: BaseApi/Transformer/AbstractTransformer.php [hunk 1]
```diff
@@ -81,6 +83,28 @@ protected function hasGroup($group)
      *
      * @return FacadeInterface
      */
+    public function cacheTransform($mixed)
+    {
+        $isObject = is_object($mixed);
```
Comment (alavieille) [inline BaseApi/Transformer/AbstractTransformer.php]: if isn't a object you can test here if not return $this->transform($mixed); and you can remove other if

File: BaseApi/Transformer/AbstractTransformer.php [hunk 2]
```diff
@@ -81,6 +83,28 @@ protected function hasGroup($group)
      *
      * @return FacadeInterface
      */
+    public function cacheTransform($mixed)
```
Comment (alavieille) [inline BaseApi/Transformer/AbstractTransformer.php]: i m not sure is util to create a new method, may be refacto transformerManager

**Comment by veolys-nanne** [comment]
update done, @alavieille

File: BaseApi/Transformer/AbstractTransformer.php (hunk 2 above)
Comment (veolys-nanne) [reply BaseApi/Transformer/AbstractTransformer.php]: done as it to not BCBreak.

**Review by alavieille — APPROVED** [review]

File: BaseApi/Transformer/TransformerManager.php [hunk 3]
```diff
@@ -58,4 +65,47 @@ public function getGroupContext()
...
+    }
+
+    /**
+     * @param string $name
+     * @param mixed  $entity
+     * @param mixed  $params
+     *
+     * @return FacadeInterface
+     */
+    public function transform($name, $entity, $params = null) {
```
Comment (alavieille) [inline BaseApi/Transformer/TransformerManager.php]: can you type params at an array ?

**Review by alavieille — CHANGES_REQUESTED** [review]

**Comment by veolys-nanne** [comment]
update done @alavieille

File: BaseApi/Transformer/AbstractTransformer.php [hunk 4]
```diff
@@ -88,20 +88,21 @@ protected function hasGroup($group)
 
     /**
      * @param mixed $mixed
+     * @param array|null $params
      *
      * @return FacadeInterface
      */
-    public function transform($mixed)
+    public function transform($mixed, array $params = null)
```
Comment (alavieille) [inline BaseApi/Transformer/AbstractTransformer.php]: why null and not array()

**Pull request merged**